pkg/logs/input/kubernetes/convertor.go

```diff
@@ -0,0 +1,86 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package kubernetes
+
+import (
+	"bytes"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	iParser "github.com/DataDog/datadog-agent/pkg/logs/parser"
+	"regexp"
+)
+
+// Convertor for converting kubernetes log line to struct Line.
+type Convertor struct {
+	iParser.Convertor
+}
+
+// Convert validates and converts kubernetes log from byte array to struct Line.
+// Kubernetes log lines follow this pattern '<timestamp> <stream> <flag> <content>',
+// see https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/logs/logs.go
+// Example:
+// 2018-09-20T11:54:11.753589172Z stdout F This is my message
+func (c *Convertor) Convert(msg []byte, defaultPrefix iParser.Prefix) *iParser.Line {
+	components := bytes.SplitN(msg, delimiter, numberOfComponents)
```

Comment:
SplitN guarantees that max(len(components)) = 4. I think the logic below for `if len(components) > numberOfComponents-1 {` might be obsolete.